### PR TITLE
storage: tag RocksDB background IO

### PR DESCRIPTION
**Pull request by tabokie**
### What is changed and how it works?

What's Changed:

Tag RocksDB compaction and flush IO flow with event listener.

### Check List 

Tests 

- Unit test

### Release note 

No release note

### CODE REVIEW

File: components/engine_rocks/src/event_listener.rs [hunk 1]
```diff
@@ -42,6 +56,19 @@ impl rocksdb::EventListener for RocksEventListener {
...
+        }
+    }
+
+    fn on_subcompaction_begin(&self, _info: &SubcompactionJobInfo) {
+        set_io_type(IOType::Compaction);
+    }
+
+    fn on_subcompaction_completed(&self, _info: &SubcompactionJobInfo) {
+        if get_io_type() == IOType::Compaction {
+            set_io_type(IOType::Other);
```
Comment (Connor1996) [inline components/engine_rocks/src/event_listener.rs:70]: `Other` is tagged for some missing tagged threads or when an error occurs for the io. I'm not sure whether it's better to tag `Other` here.

File: components/engine_rocks/Cargo.toml [hunk 2]
```diff
@@ -48,8 +49,9 @@ protobuf = "2"
 fail = "0.4"
 
 [dependencies.rocksdb]
-git = "https://github.com/tikv/rust-rocksdb.git"
+git = "https://github.com/tabokie/rust-rocksdb.git"
```
Comment (Connor1996) [inline components/engine_rocks/Cargo.toml:52]: please change it

File: cmd/src/server.rs [hunk 3]
```diff
@@ -943,7 +950,10 @@ impl TiKVServer<RocksEngine> {
 
 impl TiKVServer<RaftLogEngine> {
     fn init_raw_engines(&mut self) -> Engines<RocksEngine, RaftLogEngine> {
-        let env = get_env(self.encryption_key_manager.clone(), None /*base_env*/).unwrap();
+        let env =
+            get_encrypted_env(self.encryption_key_manager.clone(), None /*base_env*/).unwrap();
+        let env =
+            get_inspected_env(Some(Arc::new(EngineFileSystemInspector::new())), Some(env)).unwrap();
```
Comment (Connor1996) [inline cmd/src/server.rs:956]: the inspector should be passed to limiter?

File: components/engine_rocks/src/event_listener.rs (hunk 1 above)
Comment (tabokie) [reply components/engine_rocks/src/event_listener.rs:70]: The IO type after compaction is unspecified. For instance flush can be scheduled to low-priority threads used for compaction before.

File: components/engine_rocks/Cargo.toml (hunk 2 above)
Comment (tabokie) [reply components/engine_rocks/Cargo.toml:52]: This PR currently depends on tikv/rust-rocksdb#595

File: cmd/src/server.rs (hunk 3 above)
Comment (tabokie) [reply cmd/src/server.rs:956]: Inspector is constructed from limiter, which is globally accessible.

**Comment by tabokie** [comment]
IMO we should merge this ASAP as it provides a minimal integration of trackable limiter into the system, making it possible to write unit tests without the actual implementation being merged. @yiwu-arbug @Connor1996

File: components/engine_rocks/src/event_listener.rs (hunk 1 above)
Comment (Connor1996) [reply components/engine_rocks/src/event_listener.rs:70]: I mean just keep the type, future flush or compaction will set the type at the beginning anyway.

Comment (tabokie) [reply components/engine_rocks/src/event_listener.rs:70]: True. But I still don't see a strong enough reason to make the compromise. It's just *more correct* to set IO type to unknown than leaving it potentially wrong.

Comment (Connor1996) [reply components/engine_rocks/src/event_listener.rs:70]: Alright, let's keep it

**Review by Connor1996 — APPROVED** [review]
LGTM

**Review by yiwu-arbug — COMMENTED** [review]
LGTM overall. Left comment on the rust-rocksdb PR.

**Review by yiwu-arbug — APPROVED** [review]
LGTM

**Pull request merged**